src/controllers/validate.controller.ts

```diff
@@ -0,0 +1,58 @@
+import { NextFunction, Request, Response, Router } from 'express';
+
+import { Controller } from '../interfaces';
+
+import { ParserService } from '../services/parser.service';
+
+import { ProblemException } from '../exceptions/problem.exception';
+import { prepareParserConfig, tryConvertToProblemException } from '../utils/parser';
+
+const BAD_REQUEST = 'Bad Request';
+
+/**
+ * Controller which exposes the Parser functionality, to validate the AsyncAPI document.
+ */
+export class ValidateController implements Controller {
+  public basepath = '/validate';
+
+  private parserService = new ParserService();
+
+  private async validate(req: Request, res: Response, next: NextFunction) {
+    try {
+      const asyncapi = req.body?.asyncapi;
+      if (!asyncapi) {
+        return next(new ProblemException({
+          type: 'null-or-falsey-document',
+          title: BAD_REQUEST,
+          status: 400,
+          detail: 'The "asyncapi" field in the request payload is required.',
+        }));
+      } else if (typeof asyncapi !== 'string' && typeof asyncapi !== 'object') {
+        return next(new ProblemException({
+          type: 'invalid-document-type',
+          title: BAD_REQUEST,
+          status: 400,
+          detail: 'The "asyncapi" field must be a string or object.',
+        }));
+      }
+
+      const options = prepareParserConfig(req);
+      await this.parserService.parse(asyncapi, options);
```

Comment:
You don't need to check value of `asyncapi` field and return proper problem exception, because `parser-js` under the hood makes it for you. Please check code https://github.com/asyncapi/server-api/blob/4eb6c8fa9844b3edaaa0ddf2c3e734d39023e52c/src/utils/parser.ts#L76 as you can see, we try to convert given problem from `parser-js` to ServerAPI problem, also for `null-or-falsey-document` and `invalid-document-type`. You can change these lines to:

```js
try {
  const options = prepareParserConfig(req);
  await this.parserService.parse(req.body?.asyncapi, options);
```

and everything should be fine :)

Of course then you should fix `title` properties in proper tests :)

Reply:
thanks 👍